src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertiesCollector.java

```diff
@@ -1812,7 +1812,8 @@ protected boolean _replaceCreatorProperty(List<POJOPropertyBuilder> creatorPrope
             for (int i = 0, len = creatorProperties.size(); i < len; ++i) {
                 POJOPropertyBuilder cprop = creatorProperties.get(i);
                 if (cprop != null) {
-                    if (cprop.getConstructorParameter() == ctorParam) {
```

Comment:
Question: could we move null check as the first check instead, to have:

    if (creatorProperties != null && ctorParam !=null) {

Reply:
Given it's evaluated inside the loop we can't have it first checked with createProperties. And also given cprop can be null, I can't move it inline with  `if (cprop != null)`. 

what do you think? I'm not sure if I got you correctly

Reply:
I did not mean moving `if (cprop != null)` but `if (ctorParam != null)` -- latter is not loop-dependant.